query/src/test/java/org/infinispan/query/blackbox/IndexingDuringStateTransferTest.java

```diff
@@ -214,4 +228,21 @@ private void test(Consumer<Cache<Object, Object>> op, Consumer<SearchManager> ch
    private void assertFluffyIndexed(SearchManager sm) {
       assertEquals(Collections.singletonList(FLUFFY), queryAll(sm, AnotherGrassEater.class));
    }
+
+   private void waitForIndexSynced() {
+      List<Cache<String, Object>> caches = caches();
+      caches.forEach(c -> {
+         SearchManager sm = Search.getSearchManager(c);
+         DataContainer<String, Object> dataContainer = c.getAdvancedCache().getDataContainer();
+         Map<Class<?>, AtomicInteger> countPerEntity = new HashMap<>();
+         dataContainer.forEach(e -> {
+            Class<?> entity = e.getValue().getClass();
+            countPerEntity.computeIfAbsent(entity, aClass -> new AtomicInteger(0)).incrementAndGet();
```

Comment:
I'd do it without the `AtomicInteger`

```suggestion
            countPerEntity.merge(entity, 1, Integer::sum);
```

Reply:
right, looks better your way

Reply:
changed